### PR TITLE
Update doFetch to return a promise that resolves with same value as afterFetch

### PR DESCRIPTION
**Pull request by shanecav**
Solves #146.

@jamesplease Kind-of-unrelated question: While working on this, I noticed that there are `prettier` config files, but `prettier` isn't in devDependencies, so my editor wasn't auto-formatting. So I just ran it manually, but that seemed to make a bunch of formatting changes in areas I hadn't touched, so I reverted it back. Would you be interested in a PR that adds `prettier` to devDependencies, runs it on the code base once manually, and then always runs it automatically on any files that are changed by a git commit?

### CODE REVIEW

**Comment by shanecav** [comment]
Oops, forgot to update the docs. I'll add a new commit for that in just a sec.

**Comment by jamesplease** [comment]
Thanks for the PR, @shanecav ! I'll take a look in the next few days

**Comment by shanecav** [comment]
Sounds good, thanks!

**Comment by jamesplease** [comment]
Released as 3.1.0. Thank you @shanecav !!

File: src/fetch.js [hunk 1]
```diff
@@ -150,13 +150,18 @@ export class Fetch extends React.Component {
...
+    // tl;dr, the following code should never cause a React warning or error:
     //
     // `<Fetch children={({ doFetch }) => doFetch()} />
     return new Promise(resolve => {
+      // We originally wrapped this in a setTimeout so as to avoid calls to `setState`
+      // in render, which React does not allow.
+      // However, with the recent refactor to return a Promise, the `setTimeout` is not
+      // strictly necessary. At this time, the testing setup requires the timeout remain,
+      // but it will be refactored at a future time.
       setTimeout(() => {
```
Comment (shanecav) [inline src/fetch.js]: @jamesplease My understanding was that the synchronous code in the executor function passed to `new Promise()` is run immediately & synchronously, so the `setState` call in the `fetchData` method would be called synchronously, e.g. [https://jsbin.com/kelacuweci/](https://jsbin.com/kelacuweci/). So it seemed like it still needed to live inside a `setTimeout` to guarantee it's asynchronous, but I could be missing something.

Comment (jamesplease) [reply src/fetch.js]: Good catch! That’s my mistake – I was conflating the executor function with resolving the Promise, which would have been a [microtask](https://jakearchibald.com/2015/tasks-microtasks-queues-and-schedules/).

I’ll update the code comment. Thanks @shanecav !

**Pull request merged**